syllabus/inginious_syllabus.py

```diff
@@ -461,12 +465,10 @@ def register():
 
 @app.route("/activate/<string:secret>")
 def activate_account(secret):
-    try:
-        user = User.query.filter(User.activation_secret == secret).one()
-    except NoResultFound:
+    user = find_user_from_registration_hash(secret)
```

Comment:
I think it would be better to add the e-mail in the query directly so that we do not have to browse the whole db to find the user using `find_user_from_registration_hash()` : we verify the signature, if it's OK, then just create the user